tiledb/sm/array/array_directory.h

```diff
@@ -303,13 +303,17 @@ class ArrayDirectory {
    *    [`timestamp_start`, `timestamp_end`] will be considered when
    *     fetching URIs.
    * @param mode The mode to load the array directory in.
+   * @param allow_partial_fragment_overlap If we want to allow matching
```

Comment:
It looks like the the only usage of this is "if false, `consolidation_with_timestamps_supported` returns false".  What's the connection to overlapping fragments?